core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/naming/generation/DefaultNameGenerator.java

```diff
@@ -1,30 +1,18 @@
 package uk.ac.bbsrc.tgac.miso.core.service.naming.generation;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 import uk.ac.bbsrc.tgac.miso.core.data.Nameable;
 import uk.ac.bbsrc.tgac.miso.core.exception.MisoNamingException;
 import uk.ac.bbsrc.tgac.miso.core.service.naming.DefaultMisoEntityPrefix;
-import uk.ac.bbsrc.tgac.miso.core.util.LimsUtils;
 
 public class DefaultNameGenerator implements NameGenerator<Nameable> {
 
-  private static final Logger log = LoggerFactory.getLogger(DefaultNameGenerator.class);
-
   @Override
   public String generate(Nameable nameable) throws MisoNamingException {
-    if (DefaultMisoEntityPrefix.get(nameable.getClass().getSimpleName()) == null) {
-      for (Class<?> i : LimsUtils.getAllInterfaces(nameable.getClass())) {
-        if (DefaultMisoEntityPrefix.get(i.getSimpleName()) != null) {
-          log.info("Generating name based on interface :: " + DefaultMisoEntityPrefix.get(i.getSimpleName()).name() + nameable.getId());
-          return DefaultMisoEntityPrefix.get(i.getSimpleName()).name() + nameable.getId();
-        }
+    for (DefaultMisoEntityPrefix prefix : DefaultMisoEntityPrefix.values()) {
+      if (prefix.getClass().isAssignableFrom(nameable.getClass())) {
+        return prefix.name() + nameable.getId();
```

Comment:
It would be good to throw a RuntimeException if more than one applies (`nameable` implements both sample and library for some reason)